src/txadmission.cpp

```diff
@@ -0,0 +1,1028 @@
+#include "txadmission.h"
+#include "txorphanpool.h"
+#include "consensus/tx_verify.h"
+#include "main.h" // for cs_main
+#include "respend/respenddetector.h"
+#include "unlimited.h"
+#include "util.h"
+#include "utiltime.h"
+#include "validationinterface.h"
+#include <map>
+#include <string>
+#include <vector>
+
+#include <boost/thread/thread.hpp>
+
+using namespace std;
+
+
+Snapshot txHandlerSnap;
+
+void ThreadCommitToMempool();
+void ThreadTxAdmission();
+void ProcessOrphans(std::vector<uint256> &vWorkQueue);
+
+bool CheckFinalTx(const CTransaction &tx, int flags, const Snapshot &ss);
+bool CheckSequenceLocks(const CTransaction &tx, int flags, LockPoints *lp, bool useExistingLockPoints, const Snapshot &ss);
+
+
+void StartTxAdmission(boost::thread_group &threadGroup)
+{
+    txHandlerSnap.Load(); // Get an initial view for the transaction processors
+
+    // Start incoming transaction processing threads
+    for (unsigned int i = 0; i < numTxAdmissionThreads.Value(); i++)
+    {
+        threadGroup.create_thread(boost::bind(&TraceThreads<void (*)()>, strprintf("tx%d", i), &ThreadTxAdmission));
+    }
+
+    // Start tx commitment thread
+    threadGroup.create_thread(boost::bind(&TraceThread<void (*)()>, "txcommit", &ThreadCommitToMempool));
+}
+
+void StopTxAdmission()
+{
+    cvTxInQ.notify_all();
+    cvCommitQ.notify_all();
+}
+
+void FlushTxAdmission()
+{
+    bool empty = false;
+
+    while (!empty)
+    {
+        do // give the tx processing threads a chance to run
+        {
+            {
+                LOCK(csTxInQ);
+                empty = txInQ.empty() & txDeferQ.empty();
+            }
+            if (!empty)
+                MilliSleep(100);
+        } while (!empty);
+
+        {
+            boost::unique_lock<boost::mutex> lock(csCommitQ);
+            do // wait for the commit thread to commit everything
+            {
+                cvCommitQ.timed_wait(lock, boost::posix_time::milliseconds(100));
+            } while (!txCommitQ.empty());
+        }
+
+        {  // block everything and check
+            CORRAL(txProcessingCorral, CORRAL_TX_PAUSE);
+            {
+                LOCK(csTxInQ);
+                empty = txInQ.empty() & txDeferQ.empty();
+            }
+            {
+                boost::unique_lock<boost::mutex> lock(csCommitQ);
+                empty &= txCommitQ.empty();
+            }
+        }
+    }
+}
+
+// Put the tx on the tx admission queue for processing
+void EnqueueTxForAdmission(CTxInputData &txd)
+{
+    LOCK(csTxInQ);
+    bool conflict = false;
+    for (auto inp : txd.tx->vin)
+    {
+        uint256 hash = inp.prevout.hash;
+        unsigned char* first = hash.begin();
+        *first ^= (unsigned char)(inp.prevout.n & 255);
+        if (!incomingConflicts.checkAndSet(hash))
+        {
+            conflict = true;
+            break;
+        }
+    }
+    if (!conflict)
+    {
+        //LOG(MEMPOOL, "Enqueue for processing %x\n", txd.tx->GetHash().ToString());
```

Comment:
why comment this log statement since it is conditional to having `mempool` category enabled?

Reply:
its extreme log spam.  Only useful if you are running a constrained test watching exactly what is going into the mempool, but in that case its very useful.  What we need is a telecom-style log level (like python logging.info, warning, error).  But without that, commenting out useful-but-spammy logs works.